File: Models/Movie.cs
namespace ProjWebProgramming.Models
{
    public class Movie
    {
        public Guid MovieId { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public DateTime ReleaseYear { get; set; }
        public decimal Rating { get; set; }
        public decimal MovieLength { get; set; }
        public double Price { get; set; }
        public byte[]? Image { get; set; }
        public byte[]? ImageCarousel { get; set; }
        public ICollection<Genre> Genres { get; set; } = new List<Genre>();
        public List<MovieGenre>? MovieGenres { get; set; }
        public ICollection<Actor> Actors{ get; set; } = new List<Actor>();
        public List<MovieActors>? MovieActors{ get; set; }
        public ICollection<User> Users { get; set; } = new List<User>();
        public List<Wishlist>? Wishlists { get; set; }
        public Guid DirectorId { get; set; }
        public Director? Director { get; set; }
    }
}

File: react-app/src/app/models/movie.ts
import { Actor } from "./actor";
import { Director } from "./director";
import { Genre } from "./genre";
import { User } from "./user";

export interface Movie{
    movieId: string;
    title: string;
    description: string;   
    releaseYear: string;
    rating: number;
    movieLength: number;
    image?: string;
    imageCarousel?:string;
    genres?: Genre[];
    actors?: Actor[];
    users?: User[];
    directorId: string;
    director?: Director; 
}